### PR TITLE
Fixed UserOp Identifier

### PR DESCRIPTION
**Pull request by bh2smith**
The Safe Salt Nonce is one way to identify Bitte Protocol Safes, but requires querying traces:

The Query Execution by SafeSaltNonce is computationally expensive and slow: 

https://dune.com/queries/4150748 (9 minutes on large cluster)

However, appending an Identifier to the call data bytes reduces the complexity substantially:

https://dune.com/queries/4151053 (1 minute on large cluster)

This PR Proposes using both.

Observe that the SafeSaltNonce is left configurable to end users but the UserOpIdentifier will be permanently baked into the call data of every outgoing transaction from this library.

### CODE REVIEW

File: tests/unit/utils.spec.ts
Comment (tifrel) [inline tests/unit/utils.spec.ts:1]: Why are some test files called `.spec.ts` and others `.test.ts`?

**Review by tifrel — APPROVED** [review]
I see a few console.logs in here, but I am not the biggest fan of those in library functions. I know near-api-js does it and it drives me nuts. We might want to consider guarding them behind an environment variable, e.g. if NEAR_SAFE_LOGS=1 or so.

File: tests/unit/utils.spec.ts
Comment (bh2smith) [reply tests/unit/utils.spec.ts:1]: Ahh whoops. They are both common. I just forgot to only use one.

**Comment by bh2smith** [comment]
Ya, it would be good to introduce a proper logger so those logs can be disabled.

**Pull request merged**